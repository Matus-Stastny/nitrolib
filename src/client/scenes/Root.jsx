/* @flow */
import React from "react";
import styled from "styled-components";

import Day from "../public/Day";
import Num from "../public/Num";
import Price from "../public/Price";
import Text from "../public/Text";
import Time from "../public/Time";
import main from "../styles/main";
import withBrand from "../services/brand/withBrand";

const H1 = styled.h1`
  margin-top: 0;
`;

const Div = styled.div`
  background: ${props => props.theme.primary};
`;

Div.defaultProps = {
  theme: main.theme,
};

type Props = {
  brand: typeof main,
};

const Root = (props: Props) => (
  <Div>
    <H1>{props.brand.name}</H1>
    <Text t={__("Do you even lift?")} />
    <br />
    <Day date={new Date()} />
    <br />
    <Num value={1337} />
    <br />
    <Time time={new Date()} />
    <br />
    <Price value={1337} currency="EUR" />
  </Div>
);

export default withBrand(Root);
