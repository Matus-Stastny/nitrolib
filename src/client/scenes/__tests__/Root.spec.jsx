/* @flow */
import * as React from "react";
import { shallow } from "enzyme";

import Root from "../Root";

describe("#Root", () => {
  it("should render correctly", () => {
    const wrapper = shallow(<Root />);

    expect(wrapper.getElement()).toMatchSnapshot();
  });
});
