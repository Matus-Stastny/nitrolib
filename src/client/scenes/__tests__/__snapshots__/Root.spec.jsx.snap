// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#Root should render correctly 1`] = `
<Root
  brand={
    Object {
      "name": "Reactizer",
      "theme": Object {
        "primary": "deepskyblue",
      },
    }
  }
/>
`;
